### ISSUE TITLE
datalad install gets stuck reporting  git annex init :-/

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#721]
``` bash
$> datalad -l 1 install smaug:/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000
...
2016-08-24 15:46:07,100 [INFO   ] Creating a new git repo at /home/yoh/picts/mris/GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 (install.py:677)
2016-08-24 15:46:07,100 [DEBUG  ] Git clone from smaug:/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 to /home/yoh/picts/mris/GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 (gitrepo.py:443)
2016-08-24 15:46:07,529 [DEBUG  ] Git clone completed (gitrepo.py:446)
2016-08-24 15:46:07,530 [DEBUG  ] Using existing Git repository at /home/yoh/picts/mris/GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 (gitrepo.py:475)
2016-08-24 15:46:07,530 [DEBUG  ] Using existing Git repository at /home/yoh/picts/mris/GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 (gitrepo.py:475)
2016-08-24 15:46:07,530 [INFO   ] Initializing annex repo at /home/yoh/picts/mris/GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 (install.py:681)
2016-08-24 15:46:07,531 [DEBUG  ] Using existing Git repository at /home/yoh/picts/mris/GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 (gitrepo.py:475)
2016-08-24 15:46:07,531 [Level 5] Parsed ri smaug:/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 into fields {'username': '', 'scheme': u'smaug', 'fragment': '', 'hostname': '', 'query': '', 'path': u'/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000', 'password': '', 'port': ''} (network.py:280)
2016-08-24 15:46:07,532 [Level 5] Assuming ssh style ri, adjusted: {'username': '', 'scheme': u'smaug', 'fragment': '', 'hostname': '', 'query': '', 'path': u'/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000', 'password': '', 'port': ''} (network.py:290)
2016-08-24 15:46:07,532 [Level 5] Detected ssh ri (network.py:317)
2016-08-24 15:46:07,532 [Level 5] Parsed ri smaug:/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000 into fields {'username': '', 'scheme': u'smaug', 'fragment': '', 'hostname': '', 'query': '', 'path': u'/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000', 'password': '', 'port': ''} (network.py:280)
2016-08-24 15:46:07,532 [Level 5] Assuming ssh style ri, adjusted: {'username': '', 'scheme': u'smaug', 'fragment': '', 'hostname': '', 'query': '', 'path': u'/home/yoh/mris/GA/24AUG16GA_24AUG16GA_24AUG16GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000', 'password': '', 'port': ''} (network.py:290)
2016-08-24 15:46:07,532 [Level 5] Detected ssh ri (network.py:317)
2016-08-24 15:46:07,534 [DEBUG  ] Annex repository was not yet initialized at /home/yoh/picts/mris/GA/HEAD_CLINICAL_LIBRARIES_20160824_095910_981000. Initializing ... (annexrepo.py:193)
2016-08-24 15:46:07,534 [DEBUG  ] Running: ['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'annex', 'init', '--debug'] (cmd.py:352)

```

I wonder if that is control paths for ssh to blame somehow

**Comment by hanke** [comment]
I saw the same today, but without an SSH URL. Just installing from another directory on the same machine. I actually did not get stuck, but took about 90 s to complete. I could repeat the process with another repo and go the same minute of silence. Not sure whom we are paying tribute to, but we sure do ;-)

Will adjust the title...

**Comment by bpoldrack** [comment]
@yarikoptic , @mih would you mind trying to reproduce this with current branch from PR #613 or give me a dataset, that issue happened with?

**Comment by yarikoptic** [comment]
who knows if that is related or different issue but observed similar stuckiness today

```shell
$> datalad get -J3 .
get(notneeded): /mnt/btrfs/datasets/datalad/crawl/labs/haxby/attention (dataset) [already installed]
[INFO   ] Actually getting 256 files 
Total:   0%|                                                                                                                | 0.00/6.75G [00:00<?, ?B/s]yoh@hydra's password: 
Total (255 ok out of 256)100%|█████████████████████████████████████████████████████████████████████████████████████| 6.75G/6.75G [02:20<00:00, 48.2MB/s]
                                                                                                                                                        
                                                                                                                                                        
ERROR:                                                                                                                                                  
Interrupted by user while doing magic:  [subprocess.py:_communicate_with_poll:1463]
datalad get -J3 .  59.06s user 19.18s system 2% cpu 49:01.43 total
(dev)2 10962 ->1.....................................:Tue 20 Jun 2017 01:36:16 PM EDT:.
(git)smaug:/mnt/btrfs/datasets/datalad/crawl/labs/haxby/attention[master]git
$> datalad get -J3 .
get(notneeded): /mnt/btrfs/datasets/datalad/crawl/labs/haxby/attention (dataset) [already installed]
get(notneeded): /mnt/btrfs/datasets/datalad/crawl/labs/haxby/attention (directory) [nothing to get from /mnt/btrfs/datasets/datalad/crawl/labs/haxby/attention]
action summary:
  get (notneeded: 2)
(dev)2 10962.....................................:Tue 20 Jun 2017 01:36:18 PM EDT:.
(git)smaug:/mnt/btrfs/datasets/datalad/crawl/labs/haxby/attention[master]git
$> datalad --version
datalad 0.6.0.dev47
```

note that total progress bar reported 0% done! weird

**Comment by yarikoptic** [comment]
seems to be reproducible!  I have removed active sockets (with already existing ones, doesn't stall), and then it asked for password in install and get calls, and got stuck with 0% reported in global progress... I guess some interaction with recent annex enhancements to provide a single authentication prompt
```shell
(git)smaug:/mnt/btrfs/scrap/datalad/tmp/bu[master]
$> rm ~/.cache/datalad/sockets/*
zsh: sure you want to delete all the files in /home/yoh/.cache/datalad/sockets [yn]? y
(dev)2 10977.....................................:Tue 20 Jun 2017 02:05:50 PM EDT:.
(git)smaug:/mnt/btrfs/scrap/datalad/tmp/bu[master]
$> datalad install -s hydra:/data/attention/raw_bids -d . attention
[INFO   ] Cloning dataset from 'hydra:/data/attention/raw_bids' (trying 1 location candidate(s)) to '/mnt/btrfs/scrap/datalad/tmp/bu/attention' 
yoh@hydra's password: 
add(ok): attention (dataset) [added new subdataset]
add(notneeded): attention (directory) [nothing to add from /mnt/btrfs/scrap/datalad/tmp/bu/attention]                                                   
add(notneeded): .gitmodules (file) [already included in the dataset]
save(ok): /mnt/btrfs/scrap/datalad/tmp/bu (dataset)                                                                                                     
install(ok): attention (dataset)
action summary:
  add (notneeded: 2, ok: 1)
  install (ok: 1)
  save (ok: 1)
datalad install -s hydra:/data/attention/raw_bids -d . attention  9.75s user 2.87s system 68% cpu 18.527 total
(dev)2 10978.....................................:Tue 20 Jun 2017 02:06:12 PM EDT:.
(git)smaug:/mnt/btrfs/scrap/datalad/tmp/bu[master]
$> cd attention 
README  dataset_description.json  sourcedata/  sub-rid000001/  sub-rid000017/  sub-rid000027/  sub-rid000032/  sub-rid000034/  sub-rid000037/
code/   participants.tsv          stimuli/     sub-rid000012/  sub-rid000024/  sub-rid000031/  sub-rid000033/  sub-rid000036/  sub-rid000041/
(dev)2 10979.....................................:Tue 20 Jun 2017 02:06:16 PM EDT:.
(git)smaug:/mnt/btrfs/scrap/datalad/tmp/bu/attention[master]git
$> datalad get -J3 sub-rid000001 
[INFO   ] Actually getting 14 files 
Total:   0%|                                                                                                                 | 0.00/375M [00:00<?, ?B/s]yoh@hydra's password: 
Total (13 ok out of 14)100%|█████████████████████████████████████████████████████████████████████████████████████████| 375M/375M [00:12<00:00, 30.8MB/s]ERROR:                                                                                                                                                  
Interrupted by user while doing magic:  [subprocess.py:_communicate_with_poll:1463]                                         | 0.00/36.5M [00:00<?, ?B/s]
(dev)2 10980 ->1.....................................:Tue 20 Jun 2017 02:06:53 PM EDT:.
(git)smaug:/mnt/btrfs/scrap/datalad/tmp/bu/attention[master]git
$> datalad get -J3 sub-rid000001
get(notneeded): /mnt/btrfs/scrap/datalad/tmp/bu/attention/sub-rid000001 (directory) [nothing to get from /mnt/btrfs/scrap/datalad/tmp/bu/attention/sub-rid000001]
```
@mih -- something you might observe I guess as well ;)

**Comment by yarikoptic** [comment]
I wonder if that is due to
```python
        # TODO:  having two PIPEs is dangerous, and leads to lock downs so we
        # would need either threaded solution as in .communicate or just allow
        # only one to be monitored and another one just being dumped into a file
        outputstream = subprocess.PIPE if log_stdout else sys.stdout
        errstream = subprocess.PIPE if log_stderr else sys.stderr
```
which we somewhat ignored/didn't handle so far, so it also stucks on get if too much of stderr output starts to come out of git-annex

**Comment by mih** [comment]
FTR: I have seen git annex init get stuck on its own.

**Comment by yarikoptic** [comment]
I think it might have been that inability to handle both large stdout and stderr outputs... fixed now as of 0.9.2, so closing
